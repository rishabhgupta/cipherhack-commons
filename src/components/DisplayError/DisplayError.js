import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

const Error = styled.div`
    padding-top: 45px;
    color: hotpink;
    position: relative;
    vertical-align: middle;
    margin: 0;
    left: 0;
    top: 0;
    text-align: center;
`;

const Button = styled.button`
    padding: 5px 5px;
`;

/** Used to display error within a component or a page */
const DisplayError = ({
    imgSrc, errorHeading, errorSubHeading, clickAction,
}) => (
    <div id="moe-page-loader-wrapper" className="moe-semantic" style={{ height: '300px', position: 'relative' }}>
        <div style={{
            width: '100%', margin: 'auto', top: '5%', left: '0', position: 'absolute',
        }}
        >
            <Error>
                <img src={imgSrc} alt="Error in loading" />
                <h1 style={{ fontWeight: 'bold', color: '#6f6d6d' }}>{errorHeading}</h1>
                <h4 className="m-t-10" style={{ color: '#868585' }}>{errorSubHeading}</h4>
                { clickAction && <Button type="button" className="m-t-25 btn btn-primary" onClick={clickAction}>Reload</Button> }
            </Error>
        </div>
    </div>
);

DisplayError.defaultProps = {
    clickAction: false,
};

DisplayError.propTypes = {
    /** image src link for the icon */
    imgSrc: PropTypes.string.isRequired,

    /** Main heading text */
    errorHeading: PropTypes.string.isRequired,

    /** Sub heading text */
    errorSubHeading: PropTypes.string.isRequired,

    /** Call to action */
    clickAction: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
};

export default DisplayError;
q
