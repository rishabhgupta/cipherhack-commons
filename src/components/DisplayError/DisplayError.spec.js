import React from 'react';
import renderer from 'react-test-renderer';
import DisplayError from './DisplayError';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('hides reload button bar by default', () => {
    const tree = renderer.create(<DisplayError errorHeading="Alert" errorSubHeading="Something went wrong." imgSrc=""/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('toggles reload button when cta is passed', () => {
    const wrapper = shallow(
        <DisplayError
            errorHeading="Alert"
            errorSubHeading="Something went wrong."
            imgSrc=""
            clickAction={() => console.log('Called')}
        />
    );
    expect(wrapper.find({type: 'button'})).toHaveLength(1);
});
