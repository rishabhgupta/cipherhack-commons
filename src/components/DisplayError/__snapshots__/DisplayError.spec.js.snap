// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides reload button bar by default 1`] = `
<div
  className="moe-semantic"
  id="moe-page-loader-wrapper"
  style={
    Object {
      "height": "300px",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "left": "0",
        "margin": "auto",
        "position": "absolute",
        "top": "5%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-w43jqz"
    >
      <img
        alt="Error in loading"
        src=""
      />
      <h1
        style={
          Object {
            "color": "#6f6d6d",
            "fontWeight": "bold",
          }
        }
      >
        Alert
      </h1>
      <h4
        className="m-t-10"
        style={
          Object {
            "color": "#868585",
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
  </div>
</div>
`;
